### PR TITLE
Refactor for window left-align tabs (issue 495)

### PR DESCRIPTION
**Pull request by DisquietingFridge**
Changes to BuildVehicle file (or any other widget) can be omitted from merge as desired, per the notes on issue 495 suggesting BuildVehicle requires special consideration.

### CODE REVIEW

File: src/OpenLoco/Widget.h [hunk 1]
```diff
@@ -40,6 +40,10 @@ namespace OpenLoco::Ui
         static void drawViewportCentreButton(Gfx::Context* context, const Window* window, const WidgetIndex_t widgetIndex);
         static void drawTab(Window* w, Gfx::Context* ctx, int32_t imageId, WidgetIndex_t index);
 
+        //typical tab width, to be used in most (all?) cases
+        static const uint16_t defaultTabWidth = 30;
```
Comment (duncanspumpkin) [inline src/OpenLoco/Widget.h:44]: Make this a constexpr and call it kDefaultTabWidth

File: src/OpenLoco/Widget.cpp [hunk 2]
```diff
@@ -996,4 +996,25 @@ namespace OpenLoco::Ui
...
+    void Widget::leftAlignTabs(Window* const window, uint8_t firstTabIndex, uint8_t lastTabIndex, uint16_t tabWidth)
+    {
+        int16_t xPos = window->widgets[firstTabIndex].left;
+        for (uint8_t i = firstTabIndex; i <= lastTabIndex; i++)
+        {
+            if (window->isDisabled(i))
+            {
+                window->widgets[i].type = WidgetType::none;
+            }
+
```
Comment (duncanspumpkin) [inline src/OpenLoco/Widget.cpp:1009]: Could you remove this space

File: src/OpenLoco/Widget.cpp [hunk 3]
```diff
@@ -996,4 +996,25 @@ namespace OpenLoco::Ui
             Gfx::drawRect(*ctx, pos.x, pos.y + 26, 31, 1, Colours::getShade(w->getColour(WindowColour::secondary).c(), 7));
         }
     }
+
+    void Widget::leftAlignTabs(Window* const window, uint8_t firstTabIndex, uint8_t lastTabIndex, uint16_t tabWidth)
```
Comment (duncanspumpkin) [inline src/OpenLoco/Widget.cpp:1000]: I'd prefer to pass by reference if you are not handling nullptrs.

**Comment by duncanspumpkin** [comment]
Looks good. I've not tested it though.

**Comment by duncanspumpkin** [comment]
Could you address the CI issues with formatting.

File: src/OpenLoco/Widget.cpp [hunk 4]
```diff
@@ -996,4 +996,19 @@ namespace OpenLoco::Ui
...
     }
+
+    void Widget::leftAlignTabs(Window& window, uint8_t firstTabIndex, uint8_t lastTabIndex, uint16_t tabWidth)
+    {
+        int16_t xPos = window.widgets[firstTabIndex].left;
+        for (uint8_t i = firstTabIndex; i <= lastTabIndex; i++)
+        {
+            if (window.isDisabled(i) == false)
+            {
+                window.widgets[i].type = WidgetType::tab;
```
Comment (duncanspumpkin) [inline src/OpenLoco/Widget.cpp:1007]: I think as this is required we should do the inverse as well and set it to type none for when the tab is disabled.

File: src/OpenLoco/Widget.h [hunk 5]
```diff
@@ -40,6 +40,10 @@ namespace OpenLoco::Ui
         static void drawViewportCentreButton(Gfx::Context* context, const Window* window, const WidgetIndex_t widgetIndex);
         static void drawTab(Window* w, Gfx::Context* ctx, int32_t imageId, WidgetIndex_t index);
 
+        // typical tab width, to be used in most (all?) cases
+        static constexpr uint16_t kDefaultTabWidth = 30;
+        static void leftAlignTabs(Window& window, uint8_t firstTabIndex, uint8_t lastTabIndex, uint16_t tabWidth);
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Widget.h:45]: We might want to make `kDefaultTabWidth` the actual default, so we don't have to pass it in every call, e.g.:
```suggestion
        static constexpr uint16_t kDefaultTabWidth = 30;
        static void leftAlignTabs(Window& window, uint8_t firstTabIndex, uint8_t lastTabIndex, uint16_t tabWidth = kDefaultTabWidth);
```

Comment (DisquietingFridge) [reply src/OpenLoco/Widget.h:45]: Did consider this, and am fine with either- the counterpoint as suggested by LeftOfZen is that having to pass the default makes the behavior more explicit and clear.

File: src/OpenLoco/Widget.cpp [hunk 6]
```diff
@@ -996,4 +996,24 @@ namespace OpenLoco::Ui
             Gfx::drawRect(*ctx, pos.x, pos.y + 26, 31, 1, Colours::getShade(w->getColour(WindowColour::secondary).c(), 7));
         }
     }
+
+    void Widget::leftAlignTabs(Window& window, uint8_t firstTabIndex, uint8_t lastTabIndex, uint16_t tabWidth)
+    {
+        int16_t xPos = window.widgets[firstTabIndex].left;
```
Comment (LeftofZen) [inline src/OpenLoco/Widget.cpp:1002]: nit: could use `auto` here, and in the for loop below

File: src/OpenLoco/Widget.h (hunk 5 above)
Comment (LeftofZen) [reply src/OpenLoco/Widget.h:45]: yeah I initially suggested not defaulting the param when we were discussing it, but it is used quite a lot here so defaulting it does make sense as well. I don't mind either way if we default it or not

**Comment by LeftofZen** [comment]
net -201 LoC, nice!

**Review by duncanspumpkin — APPROVED** [review]

**Review by AaronVanGeffen — APPROVED** [review]

**Pull request merged**